pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -75,6 +83,44 @@ func DiffList(olds, news []Manifest, opts ...diff.Option) (*DiffListResult, erro
 	return cr, nil
 }
 
+func mergeDataAndStringData(in *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

Comment:
> This merging way will show a wrong result on the web in case there are some diffs.

I mean that the result will point to a wrong path of the field.

Reply:
I see. It is hard to have a perfect solution in this case. 🤔 

How about changing this function to 

``` golang
func normalizeNewSecret(oldSecret, newSecret Manifest) (Manifest, error) {
    var o, n v1.Secret
    runtime.DefaultUnstructuredConverter.FromUnstructured(oldSecret.Object, &o)
    runtime.DefaultUnstructuredConverter.FromUnstructured(newSecret, &n)
    
    // Move as much as possible fields from `n.Data` to `n.StringData` to make `n` close to `o` to minimize the diff.
    for k, v := n.Data {
        // Skip if the field also exists in StringData.
        if _, ok := n.StringData[v]; ok {
            continue
        }
        if _, ok := o.StringData; !ok {
            continue
        }
        // If the field is existing in `o.StringData`, we should move that field from `n.Data` to `n.StringData`
        n.StringData = base64_decoded(v)
        delete(n.Data, k)
    }
    return &unstructured.Unstructured{Object: newObj}
}

Reply:
Thank you! I think this is very good.
I change to apply this suggestion. PTAL